kagome_lattice.py

```diff
@@ -0,0 +1,190 @@
+######## Generate a Kagome Lattice #########
+
+import sys
+import numpy as np
+import matplotlib.pyplot as plt
+import yaml
+from lattice import Lattice, Site
+import argparse
+import textwrap
+
+# kagome lattice unit cell has three particles let them be  at [a(0,0), b(1,0), c(0.5,0.866)]
+# basis vectors [(1,0), (0.5,0.866)]
+# creates only 2-d lattice
+
+
+def _define_parser():
+    "Define command line argument parser for KagomeLatticeGen."
+    parser = argparse.ArgumentParser(prog="latgraph --generate kagome",
+                                     formatter_class=argparse.RawDescriptionHelpFormatter,
+                                     description=textwrap.dedent("""\
+                Generate a kagome lattice.
+                                     """))
+    parser.add_argument("cols", type=int, help="Number of unit cell per row.")
+    parser.add_argument("rows", type=int, help="Number of stacked rows.")
+    parser.add_argument("boundary_condition", type=int, choices=(1, 0),
+                        help="Boundary condition --periodic=1,open=0")
+    #parser.add_argument("--spacing", type=float, default=1.0, help="Lattice spacing.")
```

Comment:
Remove?

Reply:
Tom wanted the script to accept the size as input

Reply:
Then you'll have to use them when constructing the lattice.

Reply:
Will that be difficult to do? If so, I shall ask Tom if it is necessary.

Reply:
I don't know. That depends on your algorithm. But in principle it should just be a matter of adjusting the basis vectors.

Reply:
Since you commented out the 'spacing' argument, I assumed that you didn't care about it. And as far as I understand your code, it only uses a fixed lattice spacing. But if Tom wants it to be adjustable, you should re-enable the argument and use it in `position`.

Reply:
I have added the spacing option as well and updated it in the position